DataCleaningScripts/update_ndvi.R

```diff
@@ -0,0 +1,165 @@
+#'              corners of Portal: NW = 31.939568, -109.083177
+#'                    NE = 31.939569, -109.077460
+#'                    SW = 31.935860, -109.082948
+#'                    SE = 31.935948, -109.077317
+#' 
+#' @param centroid vector of coordinates in lon/lat of center of box [lon,lat]
+#' @param radius desired radius, in m
+#' 
+#' @return portal_area is a SpatialPolygons object that can be used for cropping rasters
+#' 
+#' @example create_portal_area(centroid = c(-109.08029, 31.937769), radius = 1125)
+#' 
+create_portal_area <- function(centroid = c(-109.08029, 31.937769), 
+                             radius = 1000) {
+  
+  center <- sf::st_sfc(sf::st_point(centroid),crs="WGS84")
+  #transform to NAD83(NSRS2007)/California Albers
+  center_transform <- sf::st_as_sf(center) %>% sf::st_transform(3488) 
+  portal_area_transform <- as(sf::st_buffer(center_transform, 1000), 'Spatial')
+  portal_area <- sp::spTransform(portal_area_transform, sp::CRS("+proj=utm +zone=12 
+                     +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0 "))
+  return(portal_area)
+  
+}
+
+
+#' Fetch new ndvi and put in new data order via getSpatialData
+#' 
+#' @param mindate minimum date to collect
+#' @param maxdate maximum date to collect
+#' @param targetpath path for data downloads 
+#'
+
+new_records <- function(mindate, maxdate, targetpath = tempdir()) {
+
+getSpatialData::set_archive(targetpath)
+getSpatialData::set_aoi(create_portal_area())
+getSpatialData::login_USGS(username = "weecology", password = Sys.getenv("USGS_PASSWORD"))
+
+records <- getSpatialData::get_records(time_range = c(mindate, maxdate),
+                                       products = "LANDSAT_8_C1")
+records <- records[records$level == "sr_ndvi",]
+records <- getSpatialData::check_availability(records)
+records <- getSpatialData::get_data(records)
+
+# Submit order for any new records not downloaded
+records <- getSpatialData::order_data(records)
+
+return(records)
+}
+
+#' @title extract and mask raster
+#' @description extracts raster files from compressed files, crops to Portal, 
+#' and masks with pixel_qa layer
+#'
+#' @param records new records to be used
+#' @param targetpath path where the cropped and masked rasters are to be stored
+#'
+extract_and_mask_raster <- function(records, targetpath = tempdir()) {
+  tarfile <- as.character(unlist(records["dataset_file"]))[1]
+  record_id <- as.character(unlist(records["record_id"]))[1]
+  # extract ndvi and pixel_qa rasters
+  untar(tarfile, exdir = targetpath)
+  # create portal box object for cropping
+  portal_area <- create_portal_area()
+  
+  # read in raster files; crop to portal_box; delete full-size raster files  
+  r <- raster::raster(paste0(targetpath,"/", record_id, "_pixel_qa.tif"))
+  scene <- raster::crop(r,portal_area)
+  
+  m <- raster::raster(paste0(targetpath,"/", record_id, "_sr_ndvi.tif"))
+  pixelqa <- raster::crop(m,portal_area)
```

Comment:
Somehow these are reversed compared to the old code. It matters when you get to raster::mask() -- as written it was masking the pixel_qa layer with the ndvi layer instead of the other way around, which would explain the nonsense results.
```suggestion
  r <- raster::raster(paste0(targetpath,"/", record_id, "_sr_ndvi.tif"))
  scene <- raster::crop(r,portal_area)
  
  m <- raster::raster(paste0(targetpath,"/", record_id, "_pixel_qa.tif"))
  pixelqa <- raster::crop(m,portal_area)
```